test/crc32_test.c

```diff
@@ -0,0 +1,208 @@
+/* crc32_test.c -- crc32 unit test
+ * Copyright (C) 2019-2021 IBM Corporation
+ * Authors: Rogerio Alves    <rogealve@br.ibm.com>
+ *          Matheus Castanho <msc@linux.ibm.com>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+typedef unsigned long crc32_type;
+typedef unsigned char buf_type;
+typedef unsigned int len_type;
+#  define CRC_FMT "%08lX"
```

Comment:
I wonder if this `CRC_FMT` should be moved into zlib-ng.h. @Dead2 looks like z_crc_t was removed in https://github.com/zlib-ng/zlib-ng/commit/4db4cfdb5badc8860f7410732b12c45216d709b3. Do we need to bring it back for zlib.h ZLIB_COMAT?

Reply:
All optimized crc32 functions should use only `uint32_t` as CRC32 variable type, the conversion for ZLIB_COMPAT API is done in `crc32_z()` and `crc32()` inside crc32.c. All length parameters should always be `size_t`. Conversion from `unsigned int` is also done in crc32.c.

Reply:
get_crc_table may have produced unsigned long return type in zlib.

Reply:
If the test is statically linked to zlib-ng, it can access functable directly, and such there will be no differences... For dynamic linking, simple cast when calling `crc32_z` is enough. I pointed out to how it is done in crc32.c, because essentially what is needed is same backwards. It would be a lot easier, if there would be "undocumented API" to access functable when building against shared version of zlib-ng.